src/connector/src/parser/scalar_adapter.rs

```diff
@@ -157,80 +176,142 @@ impl ScalarAdapter<'_> {
             (ScalarRefImpl::Utf8(s), _, Kind::Enum(_)) => {
                 ScalarAdapter::Enum(EnumString(s.to_owned()))
             }
-            _ => ScalarAdapter::Builtin(scalar),
+            (ScalarRefImpl::List(list), &Type::NUMERIC_ARRAY, _) => {
+                let mut vec = vec![];
+                for scalar in list.iter() {
+                    vec.push(match scalar {
```

Comment:
The `scalar` should rename to `datum`, because `Datum` is `Option<ScalarImpl>`. And I am curious why not `vec.push( datum.map(|scalar|  ScalarAdapter::from_scalar(scalar, &Type::NUMERIC) ) )` to call the `from_scalar` recursively? Just add one more branch to handle `(ScalarRefImpl::Decimal, &Type::NUMERIC)` as base case.

Reply:
> The scalar should rename to datum, because Datum is Option<ScalarImpl>

updated.

> And I am curious why not vec.push( datum.map(|scalar|  ScalarAdapter::from_scalar(scalar, &Type::NUMERIC) ) ) to call the from_scalar recursively? Just add one more branch to handle (ScalarRefImpl::Decimal, &Type::NUMERIC) as base case.

For two reason:
1. `from_scalar ` return `ScalarAdapter` directly, but `ScalarAdapter::NumericList` requires `Vec<Option<PgNumeric>>` actually. 
2. If we don't handle `(ScalarRefImpl::Decimal, &Type::NUMERIC)`, `ScalarRefImpl::Decimal` will be `builtin`. We have implement `ToSql` for rw's `Decimal`, so it acceptale by PG. If we handle `(ScalarRefImpl::Decimal, &Type::NUMERIC)`, we have to first convert it to string and then parse as `PgNumeric`, seems unnecessary.

Reply:
> If we handle (ScalarRefImpl::Decimal, &Type::NUMERIC), we have to first convert it to string and then parse as PgNumeric, seems unnecessary.

Your are right, I got it.